internal/backends/testing/search_api.rs

```diff
@@ -51,11 +59,98 @@ impl ElementHandle {
+        let result =
+            search_item(&item_tree, |elem| elem.accessible_label().is_some_and(|x| x == label));
+        result.into_iter()
+    }
+
+    /// This function searches through the entire tree of elements of this window and looks for
+    /// elements by their id. The id is a qualified string consisting of the name of the component
+    /// and the assigned name within the component. In the following examples, the first Button
+    /// has the id "MyView::submit-button" and the second button "App::close":
+    ///
+    /// ```slint,no-preview
+    /// component MyView {
+    ///    submit-button := Button {}
+    /// }
+    /// export component App {
+    ///     VerticalLayout {
+    ///         close := Button {}
+    ///     }
+    /// }
+    /// ```
+    pub fn find_by_element_id(
+        component: &impl i_slint_core::api::ComponentHandle,
+        id: &str,
+    ) -> impl Iterator<Item = Self> {
+        // dirty way to get the ItemTreeRc:
+        let item_tree = WindowInner::from_pub(component.window()).component();
+        let result = search_item(&item_tree, |elem| {
+            elem.element_type_names_and_ids().unwrap().any(|(_, eid)| eid == id)
         });
-        result.into_iter().map(ElementHandle)
+        result.into_iter()
+    }
+
+    /// This function searches through the entire tree of elements of `component`, looks for
+    /// elements with given type name.
+    pub fn find_by_element_type_name(
+        component: &impl i_slint_core::api::ComponentHandle,
+        type_name: &str,
+    ) -> impl Iterator<Item = Self> {
+        // dirty way to get the ItemTreeRc:
+        let item_tree = WindowInner::from_pub(component.window()).component();
+        let result = search_item(&item_tree, |elem| {
+            elem.element_type_names_and_ids().unwrap().any(|(tn, _)| tn == type_name)
+        });
+        result.into_iter()
+    }
+
+    /// Returns an iterator over tuples of element type names and their ids. Returns None if the
+    /// element is not valid anymore.
+    ///
+    /// Elements can have multiple type names and ids, due to inheritance.
+    /// In the following example, the `PushButton` element returns for `element_type_names_and_ids`
+    /// the following tuples:
+    /// entries:
+    ///   * ("PushButton", "App::mybutton")
+    ///   * ("ButtonBase", "PushButton::root")
+    ///   * ("", "ButtonBase::root")
+    ///
+    /// ```slint,no-preview
+    /// component ButtonBase {
+    ///    // ...
+    /// }
+    /// component PushButton inherits ButtonBase {
+    /// }
+    /// export component App {
+    ///     mybutton := PushButton {}
+    /// }
+    /// ```
+    ///
+    /// ```rust
+    /// # i_slint_backend_testing::init_no_event_loop();
+    /// # slint::slint!{ #[debug_info]
+    /// # component ButtonBase { }
+    /// # component PushButton inherits ButtonBase { }
+    /// # export component App {
+    /// #    mybutton := PushButton {}
+    /// # }
+    /// # }
```

Comment:
why not put the actual slint! macro there in the doc instead of the slint code separately?
If you keep the slint! macro visible in the doc example, it makes it more accurate and the code example can't become stale.

Reply:
Good point, I can try that.